packages/calcite-components/src/components/input-number/input-number.e2e.ts

```diff
@@ -715,6 +715,21 @@ describe("calcite-input-number", () => {
       const input = await page.find("calcite-input-number >>> input");
       expect(await input.getProperty("value")).toBe("2");
     });
+
+    it("allows trailing decimal separator", async () => {
+      const page = await newE2EPage();
+      await page.setContent(html`<calcite-input value="1.1"></calcite-input>`);
+      const calciteInput = await page.find("calcite-input");
+      const inputEventSpy = await calciteInput.spyOnEvent("calciteInputInput");
+
+      await calciteInput.click();
+      await page.keyboard.press("Backspace");
+      await page.waitForChanges();
+      expect(inputEventSpy).toHaveReceivedEvent();
+      const input = await page.find("calcite-input >>> input");
+
+      expect(await input.getProperty("value")).toBe("1.");
```

Comment:
the component's value should still be `1`. You will need to go into the shadowRoot and check the value of the internal input to verify it is `1.`

Reply:
Nevermind that is what you were doing it was other input tests failing. I need read closer before opening my big mouth lol

Reply:
no problem. The issues with other tests is the immutation of `localizedValue` when the user tries to delete the decimal part.  The logic applies even if the user is typing out a value like `0.05`. By the time user types `0.0` the validation is happening hence changing the localizedValue to `0` again. Current setup avoids this by not changing the `localizedValue` variable until the user inputs a valid value which allows users to type the value of their without any auto correction.  

Trying to figure out a reliable way to identify if the user is adding a value in the input field or deleting the value so that we can apply the logic only for deletion case.

Reply:
Gotcha. 

> Trying to figure out a reliable way to identify if the user is adding a value in the input field or deleting the value so that we can apply the logic only for deletion case.

Some potential options for investigation:

1. only sanitzing the trailing decimal if the length of the previous value is more than the current one
2. only sanitzing the trailing decimal on backspace and delete key events. And the increment down click event too probably

Reply:
yeah i was using the second option which is not feasible with `inputEvent` . i had to create a combination of above. Realized there was an another bug with the input when the user has leading `0`'s in the decimal. If the user deletes the `2` in `0.00002` then the value becomes `0` which is similar to #7039 . I tried adding those trailing zero's back in the decimal but the localize method is removing them too. We may have to follow similar technique of adding back those zeros like we did for decimal separator.

Reply:
or we can refactor [this](https://github.com/Esri/calcite-components/blob/master/packages/calcite-components/src/utils/number.ts#L47) method.

Reply:
Yeah the input lifecycle methods will probably need a refactor. Once you start doing crazy stuff like this to fix bugs you know you've gone too far lol.